data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/ScanObjectWorker.java

```diff
@@ -0,0 +1,187 @@
+     * based on s3 select configuration provided.
+     */
+    @Override
+    public void run() {
+        scanOptionsBuilderList.forEach(this::parseS3ScanObjects);
+    }
+
+    /**
+     * Method will parse the s3 object and write to {@link Buffer}
+     */
+    void parseS3ScanObjects(ScanOptionsBuilder scanOptionsBuilder) {
+        final S3ObjectRequest.Builder s3ObjectRequestBuilder = new S3ObjectRequest.Builder(buffer, numberOfRecordsToAccumulate,
+                bufferTimeout, s3ObjectPluginMetrics);
+        final S3ObjectHandler s3ObjectHandler;
+        if(scanOptionsBuilder.getExpression() != null){
+            S3ObjectRequest s3ObjectRequest = s3ObjectRequestBuilder.expression(scanOptionsBuilder.getExpression())
+                    .serializationFormatOption(scanOptionsBuilder.getSerializationFormatOption())
+                    .expressionType(scanOptionsBuilder.getExpressionType())
+                    .s3AsyncClient(s3AsyncClient).compressionType(scanOptionsBuilder.getCompressionType())
+                    .s3SelectResponseHandler(new S3SelectResponseHandler()).eventConsumer(eventMetadataModifier)
+                    .s3SelectCSVOption(scanOptionsBuilder.getS3SelectCSVOption())
+                    .s3SelectJsonOption(scanOptionsBuilder.getS3SelectJsonOption()).build();
+            s3ObjectHandler = new S3SelectObjectWorker(s3ObjectRequest);
+        } else {
+            final S3ObjectRequest s3ObjectRequest = s3ObjectRequestBuilder.bucketOwnerProvider(bucketOwnerProvider)
+                    .eventConsumer(eventMetadataModifier).codec(scanOptionsBuilder.getCodec()).s3Client(s3Client)
+                    .compressionEngine(scanOptionsBuilder.getCompressionOption().getEngine()).build();
+            s3ObjectHandler = new S3ObjectWorker(s3ObjectRequest);
+        }
+        scanOptionsBuilder.getKeys().forEach(key ->{
+            final S3ObjectReference s3ObjectReference = S3ObjectReference.bucketAndKey(scanOptionsBuilder.getBucket(), key).build();
+            processS3ObjectKeys(s3ObjectReference,s3ObjectHandler, scanOptionsBuilder);
+        });
+    }
+
+
+    private void processS3ObjectKeys(final S3ObjectReference s3ObjectReference, final S3ObjectHandler s3ObjectHandler, final ScanOptionsBuilder scanOptionsBuilder){
+        final S3ObjectDetails s3ObjDetails = getS3ObjectKeyLastModifiedTime(s3ObjectReference);
+        final boolean isKeyMatchedBetweenTimeRange = isKeyMatchedBetweenTimeRange(s3ObjDetails.getS3ObjectLastModifiedTimestamp(),
+                scanOptionsBuilder.getRange(),
+                LocalDateTime.parse(scanOptionsBuilder.getStartDate()));
+        if(isKeyMatchedBetweenTimeRange && (isKeyProcessedByS3Scan(s3ObjDetails))){
+            updateKeyProcessedByS3Scan(s3ObjDetails);
+            try{
+                s3ObjectHandler.parseS3Object(s3ObjectReference);
+            }catch (IOException ex){
```

Comment:
Thanks for your review Dinu John,
@dlvenable can you please suggest here.

Reply:
We are swallowing the exception in the catch block and it's not thrown. What should be the user action when they see this error log ?

Reply:
Thanks for your review Dinu, Please suggest what should be the action in case of any Exception in ScanObjectWorker. for now as per requirement we are just loging those exceptions and continuing with the next object scan.

Reply:
David/Ashok are we good with the requirement here ? I think we should add a metric and provide user the visibility.